cfme/tests/automate/custom_button/test_buttons.py

```diff
@@ -589,7 +591,38 @@ def test_attribute_override():
     Bugzilla:
         1651099
     """
-    pass
+    attributes = [
+        ("class", "Request"),
+        ("instance", "TestNotification"),
+        ("message", "digitronik_msg"),
+        ("namespace", "/System"),
+    ]
+    req = "call_instance_with_message"
+    executions = {"simulation": "create", "execution": "digitronik_msg"}
+    group = buttongroup("PROVIDER")
+    button = group.buttons.create(
+        text="btn_{}".format(fauxfactory.gen_alphanumeric(3)),
+        hover="hover_{}".format(fauxfactory.gen_alphanumeric(3)),
+        system="Request",
+        request=req,
+        attributes=attributes,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    # Check msg override for simulation and actual execution
+    for exe, msg in executions.items():
+        pattern = "[miqaedb:/System/Request/TestNotification#{msg}]".format(msg=msg)
+        log = LogValidator("/var/www/miq/vmdb/log/automation.log", matched_patterns=[pattern])
+        log.fix_before_start()
+
+        if exe == "simulation":
+            button.simulate(provider.name, request="call_instance_with_message")
+        else:
+            view = navigate_to(provider, "Details")
+            custom_button_group = Dropdown(view, group.hover)
+            custom_button_group.item_select(button.text)
+
+        log.wait_for_log_validation()
```

Comment:
You have defined `log` within the `for` loop above, and then call `wait_for_log_validation()` outside the loop.

I think you'll find, inspecting the test logs, that you're only asserting the last `msg` from the for loop.

You need to take advantage of your updates to `LogValidator` and its counting of the patterns.

You should create `LogValidator` before the loop, passing something like `matched_patterns=['[miqaedb{}]'.format(msg) for msg in executions.values()]` in order to register all of the relevant messages. Probably better to expand that statement though, so that you then have the list of expected strings to compare count against later.

Then fix the log, and iterate over `executions.keys()` to call the custom button `item_select` methods that trigger all the expected log messages.

Then use the `LogValidator` to assert that you got at least one of each message (default wait_for_log_validation behavior)

Reply:
> You have defined `log` within the `for` loop above, and then call `wait_for_log_validation()` outside the loop.

Mike `wait_for_log_validation()` is inside for loop. so its calling two times for both operation means button `execution` and `simulation`
> 
> I think you'll find, inspecting the test logs, that you're only asserting the last `msg` from the for loop.
> 
> You need to take advantage of your updates to `LogValidator` and its counting of the patterns.
> 
> You should create `LogValidator` before the loop, passing something like `matched_patterns=['[miqaedb{}]'.format(msg) for msg in executions.values()]` in order to register all of the relevant messages. Probably better to expand that statement though, so that you then have the list of expected strings to compare count against later.

I like this option; better to use `matches` property, but I need `wait_for` with it. Let's try this :+1: 
> 
> Then fix the log, and iterate over `executions.keys()` to call the custom button `item_select` methods that trigger all the expected log messages.
> 
> Then use the `LogValidator` to assert that you got at least one of each message (default wait_for_log_validation behavior)